src/libsystemd-network/dhcp-option.c

```diff
@@ -48,6 +49,32 @@ static int option_append(uint8_t options[], size_t size, size_t *offset,
                 *offset += 1;
                 break;
 
+        case SD_DHCP_OPTION_USER_CLASS: {
+                unsigned i = 0;
+                char **s;
+
+                STRV_FOREACH(s, optval) {
+                        i += strlen(*s) + 1;
+                }
+
+                if (size < *offset + i + 2)
+                        return -ENOBUFS;
+
+                options[*offset] = code;
+                options[*offset + 1] =  i;
```

Comment:
spurious space...

Also, i must be 0…255 apparently, could this pleas be checked explicitly before?

Reply:
I don't see this in RFC the length it's not mentioned . Could you please point out.

Reply:
well, options[] is an array of type uint8_t. Which means it *has* to be in the 0…255 range, but the value you assign, i.e. `i`, is of type size_t and might very well grow beyond 255, i.e. what fits in a byte. This is may easily be overflowed

Reply:
right updated.